python/revng/scripting/daemon_project.py

```diff
@@ -0,0 +1,203 @@
+
+        if not result["uploadFile"]:
+            raise RuntimeError("File upload failed")
+
+    def get_artifact(self, artifact_name: str, targets_class: List[Any] = []) -> bytes:
+        self._apply_diff()
+
+        if not targets_class:
+            targets_class = self.model.Functions
+
+        targets = [tc.key() for tc in targets_class]
+
+        query = gql(
+            """
+            query($step: String!, $paths: String!, $index: BigInt!) {
+                produceArtifacts(step: $step, paths: $paths, index: $index) {
+                    ... on Produced {
+                        result
+                    }
+                }
+            }
+        """
+        )
+        variables = {
+            "step": artifact_name,
+            "paths": ",".join(targets),
+            "index": self._get_content_commit_index(),
+        }
+
+        response = self.client.execute(query, variable_values=variables)
+        return response["produceArtifacts"]["result"].encode()
+
+    def get_artifacts(self, params: dict[str, list[Any]]) -> dict:
+        result = {}
+        for artifact_name, targets in params.items():
+            result[artifact_name] = self.get_artifact(artifact_name, targets)
+
+        return result
+
+    def analyze(self, analysis_name: str, target_class: list[Any] = []) -> Binary:
+        container_input, container_kind = self._get_analysis_inputs(analysis_name)
+        step = self._get_step_name(analysis_name)
+
+        variables = {
+            "step": step,
+            "analysis": analysis_name,
+            "index": self._get_content_commit_index(),
+        }
+        if not container_input:
+            query = gql(
+                """
+                mutation($step: String!, $analysis: String!, $index: BigInt!) {
+                    runAnalysis(step: $step, analysis: $analysis, index: $index) {
+                        __typename
+                    }
+                }
+            """
+            )
+        else:
+            query = gql(
+                """
+                mutation($step: String!, $analysis: String!, $container: String!, $index: BigInt!) {
+                    runAnalysis(step: $step, analysis: $analysis,
+                                containerToTargets: $container, index: $index) {
+                        __typename
+                    }
+                }
+            """
+            )
+            if container_kind == "root":
+                targets = [f":{container_kind}"]
+            else:
+                if not target_class:
+                    target_class = self.model.Functions
+                targets = [f"{tc.key()}:{container_kind}" for tc in target_class]
+            variables["container"] = json.dumps({container_input: targets})
+
+        self.client.execute(query, variable_values=variables)
+        model = self._get_model()
+        self.set_model(model)
+        return model
+
+    def analyses_list(self, analysis_name: str) -> Binary:
```

Comment:
I made this an private method in the `daemon` project that gets called when the user requests an analysis list, so that from the user perspective the interface is the same as the `cli` project.